AI/VCAI/Fuzzy.cpp

```diff
@@ -199,6 +202,20 @@ void FuzzyHelper::initTacticalAdvantage()
 	}
 }
 
+float FuzzyHelper::calculateTurnDistanceInputValue(const CGHeroInstance * h, int3 tile) const
+{
+	float turns = 0.0f;
+	float distance = CPathfinderHelper::getMovementCost(h, tile);

```

Comment:
Please fix it, it is wrong. The only correct way I know is
```
ui32 distanceToTile(const CGHeroInstance* hero, int3 pos) {
	auto pathsInfo = cb->getPathsInfo(hero);

	return distanceToTile(pathsInfo, pos);
}

ui32 distanceToTile(const CPathsInfo* pathsInfo, int3 pos) {
	auto pathInfo = pathsInfo->getPathInfo(pos);
	auto hero = pathsInfo->hero;
	int result = (int)(pathInfo->turns * hero->maxMovePoints(true) + hero->movement) - (int)pathInfo->moveRemains;

	return std::max(result, 0);
}
```
But maybe std:max should be replaced somehow.

Reply:
this is copypaste of thing that was initially in evaluate(VisitTile ...). I have no clue how it works and I would deal with it after merge.

Reply:
Hmm, we can fix both now or wait for Pathfinder rewrite. Either way, there are at least 2 other places in VCAI where Pathfinder doesn't perform as expected.

Reply:
Any more details about these places?

Reply:
One causes this bug: https://bugs.vcmi.eu/view.php?id=2980

The other is `int3 SectorMap::firstTileToGet()` calling `cb->buildBoat()` in a middle of logic

Reply:
So, conclusion -> wait until pathfinder gets rewrite, covering SectorMap functionality

Reply:
Unless current CPathfinderHelper::getMovementCost() call is gamebreaking?

Reply:
I think boat logic will be covered also. Pathfinder will tell that boat is needed and payh execution logic in vcai will hire a boat

Reply:
So, anyway, what is wrong with Dydzio's code and why it should be changed?